### DISCUSSION TITLE
Bootstrap CSS not working

### DISCUSSION THREAD

**Discussion by Joy-Prince** [dompdf/dompdf#3480] [Q&A]
Hi Developer,
I'm using your Dompdf in a project where I need to print the page as PDF. But when I try to do that bootstrap design didn't comes as it is. Every design became collapse .. 

$html = view()->render();

when I use above code for a particular blade file the output exactly as same as the design but 

 $dompdf = new Dompdf();
    
        // Load HTML into Dompdf
        $dompdf->loadHtml($html);
    
        // (Optional) Setup the paper size and orientation
        $dompdf->setPaper('A4', 'portrait');
    
        // Render the PDF
        $dompdf->render();
    
        // Output the generated PDF to Browser
        $dompdf->stream();

After I use the above code everything become clumsy..

Is there any solution for this 
 
The Actual Design
![FireShot Capture 028 - VIew Invoice - 127 0 0 1](https://github.com/dompdf/dompdf/assets/65761385/8d8a16a9-4293-4e17-a808-28d629c99a6c)

The design I get after download as PDF
![image](https://github.com/dompdf/dompdf/assets/65761385/3c4f1599-01be-459a-b8d3-7b51e172afaf)

**Comment by bsweeney** [comment]
There are a few features of CSS that dompdf does not support that will result in a non-optimal render. However, the collapsed layout is expected because of the page size. A4 pages are 595.28 pt wide, which may fall within 768 pixel width defined by Bootstrap depending on your page margins and Dompdf-defined DPI. The current default DPI is 96 which leads to a _page_ width equivalent to 793.71 pixels.

Would need to see your HTML to do any testing myself to determine how well the render can be improved. But for a start you can try setting the DPI to something like 150 or 300 and see how it looks. Note that the DPI will impact the rendered size any element whose measurement is defined in pixels.
